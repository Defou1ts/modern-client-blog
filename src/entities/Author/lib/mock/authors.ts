import type { AuthorWithLocales } from '../../interfaces';

export const authors: AuthorWithLocales[] = [
	{
		id: 1,
		avatarURL:
			'https://images.pexels.com/photos/7562313/pexels-photo-7562313.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Floyd',
				surname: 'Miles',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Флойд',
				surname: 'Майлз',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 2,
		avatarURL:
			'https://images.pexels.com/photos/16424676/pexels-photo-16424676.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Diana',
				surname: 'Russel',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Дианал',
				surname: 'Рассл',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 3,
		avatarURL:
			'https://images.pexels.com/photos/5704849/pexels-photo-5704849.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Jenny',
				surname: 'Wilson',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Дженни',
				surname: 'Уилсон',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 4,
		avatarURL:
			'https://images.pexels.com/photos/6205509/pexels-photo-6205509.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Leslie',
				surname: 'Alexander',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Александр',
				surname: 'Лесли',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 5,
		avatarURL:
			'https://images.pexels.com/photos/14687031/pexels-photo-14687031.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Guy',
				surname: 'Hawkins',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Гай',
				surname: 'Хавкинс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 6,
		avatarURL:
			'https://images.pexels.com/photos/4467687/pexels-photo-4467687.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Eleanor',
				surname: 'Pena',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Элеонора',
				surname: 'Пена',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 7,
		avatarURL:
			'https://images.pexels.com/photos/5262459/pexels-photo-5262459.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Robert',
				surname: 'Fox',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Роберт',
				surname: 'Фокс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 8,
		avatarURL:
			'https://images.pexels.com/photos/6147118/pexels-photo-6147118.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Jacob',
				surname: 'Jones',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Джейкоб',
				surname: 'Джонс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
];
