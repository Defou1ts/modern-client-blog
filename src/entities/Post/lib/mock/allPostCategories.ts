import BusinessIcon from '@app/assets/icons/business.png';
import StartupIcon from '@app/assets/icons/startup.png';
import EconomyIcon from '@app/assets/icons/economy.png';
import TechnologyIcon from '@app/assets/icons/technology.png';

import type { PostCategory } from '@entities/Post/interfaces';

export const allPostCategories: PostCategory[] = [
	{
		category: 'business',
		icon: BusinessIcon,
		title: {
			en: 'Business',
			ru: 'Бизнес',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'startup',
		icon: StartupIcon,
		title: {
			en: 'Startup',
			ru: 'Стартап',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'economy',
		icon: EconomyIcon,
		title: {
			en: 'Economy',
			ru: 'Экономика',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'technology',
		icon: TechnologyIcon,
		title: {
			en: 'Technology',
			ru: 'Технологии',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
];
