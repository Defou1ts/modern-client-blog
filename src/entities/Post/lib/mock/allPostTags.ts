import type { PostTag } from '../../types';

export const allPostTags: PostTag[] = ['business', 'experience', 'screen', 'technology', 'marketing', 'life'];
