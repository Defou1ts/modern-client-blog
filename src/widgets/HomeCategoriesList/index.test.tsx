import React from 'react';

import { useRouter } from 'next/router';

import renderer from 'react-test-renderer';

import { HomeCategoriesList } from './index';

import { allPostCategories } from '@entities/Post/lib/mock/allPostCategories';

import type { NextRouter } from 'next/router';

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}));

describe('HomeCategoriesList', () => {
	beforeAll(() => {
		const useRouterMock = {
			locale: 'en',
			locales: ['en', 'ru'],
			pathname: '/',
		};
		(useRouter as jest.Mock<NextRouter>).mockReturnValue(useRouterMock as NextRouter);
	});

	it('HomeCategoriesList snapshot', () => {
		const component = renderer.create(<HomeCategoriesList categories={Object.values(allPostCategories)} />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
