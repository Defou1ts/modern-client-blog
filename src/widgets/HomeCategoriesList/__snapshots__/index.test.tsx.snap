// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCategoriesList HomeCategoriesList snapshot 1`] = `
<article>
  <h2
    className="h h2 title"
  >
    home.categories-title
  </h2>
  <div
    className="list horizontal"
  >
    <a
      className=""
      href="/category/business"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="card large"
      >
        <img
          alt="Business"
          className="icon"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          title="Business"
          width={40}
        />
        <h3
          className="h h3 title"
        >
          Business
        </h3>
        <p
          className="p medium description"
        >
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
        </p>
      </div>
    </a>
    <a
      className=""
      href="/category/startup"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="card large"
      >
        <img
          alt="Startup"
          className="icon"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          title="Startup"
          width={40}
        />
        <h3
          className="h h3 title"
        >
          Startup
        </h3>
        <p
          className="p medium description"
        >
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
        </p>
      </div>
    </a>
    <a
      className=""
      href="/category/economy"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="card large"
      >
        <img
          alt="Economy"
          className="icon"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          title="Economy"
          width={40}
        />
        <h3
          className="h h3 title"
        >
          Economy
        </h3>
        <p
          className="p medium description"
        >
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
        </p>
      </div>
    </a>
    <a
      className=""
      href="/category/technology"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="card large"
      >
        <img
          alt="Technology"
          className="icon"
          data-nimg="1"
          decoding="async"
          height={40}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          title="Technology"
          width={40}
        />
        <h3
          className="h h3 title"
        >
          Technology
        </h3>
        <p
          className="p medium description"
        >
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
        </p>
      </div>
    </a>
  </div>
</article>
`;
